web/src/stream_edit.ts

```diff
@@ -99,6 +99,59 @@ export function setup_subscriptions_tab_hash(tab_key_value: string): void {
     }
 }
 
+export function open_stream_edit_modal(stream_id: number, is_archived = false): void {
```

Comment:
Feels weird to not pass `is_archived` as `true` even when the stream is archived, like we do when editing archived stream directly from its settings UI and not from stream creation form since we do not show "(archived)" in the modal heading when opening it from stream settings form.

I think we should just add "(archived)" in the modal heading for all cases when editing an archived stream. So, you can add one more preparatory commit before this commit to do that. This way we would not pass `is_archived` parameter here and instead compute it in the function itself.